### PR TITLE
Finalized support for built-in functions in the semantizer

### PR DESCRIPTION
**Pull request by MaksimShagov**

### CODE REVIEW

File: backend/src/semantizer/semantizer.cpp [hunk 1]
```diff
@@ -112,14 +112,57 @@ static BinaryOperation convertToFloatOperation(BinaryOperation operation) {
...
+        return BuiltInTypes::IntType;
+    } else if (type == NodeType::VariableName) {
+        return searchVariable(node, tables, errors);
+    } else if (type == NodeType::FunctionCall) {
+        return processFunctionCall(node->firstChild(), tables, functions, errors);
+    } else if (type == NodeType::FloatingPointLiteralValue) {
+        return BuiltInTypes::FloatType;
+    } else if (type == NodeType::IntegerLiteralValue) {
+        return BuiltInTypes::IntType;
+    }
```
Comment (vla5924) [inline backend/src/semantizer/semantizer.cpp:142]:
```suggestion
    }
    if (type == NodeType::VariableName)
        return searchVariable(node, tables, errors);
    if (type == NodeType::FunctionCall)
        return processFunctionCall(node->firstChild(), tables, functions, errors);
    if (type == NodeType::FloatingPointLiteralValue)
        return BuiltInTypes::FloatType;
    if (type == NodeType::IntegerLiteralValue)
        return BuiltInTypes::IntType;
```

**Review by arteboss — DISMISSED** [review]
KPACABA
image

File: backend/src/semantizer/semantizer.cpp [hunk 2]
```diff
@@ -112,14 +112,57 @@ static BinaryOperation convertToFloatOperation(BinaryOperation operation) {
...
+
 static TypeId processFunctionCall(Node::Ptr &node, const std::list<VariablesTable *> &tables, FunctionsTable &functions,
                                   ErrorBuffer &errors) {
     const std::string &funcName = node->firstChild()->str();
     bool isPrintFunction = (funcName == "print");
+    if (isPrintFunction) {
+        auto exprNode = node->lastChild()->lastChild();
+        auto child = exprNode->firstChild();
+        auto type = child->type;
+        exprNode->value = processPrintFunction(child, type, tables, functions,
```
Comment (vla5924) [inline backend/src/semantizer/semantizer.cpp:154]: Что за кириллица у нас тут?

File: tests/backend/semantizer.cpp [hunk 3]
```diff
@@ -44,6 +44,76 @@ TEST(Semantizer, can_fill_variables_table) {
     ASSERT_EQ(tree_str, tree.dump());
 }
 
+TEST(Semantizer, can_work_with_print_function_with_float_args) {
```
Comment (vla5924) [inline tests/backend/semantizer.cpp:47]:
```suggestion
TEST(Semantizer, can_process_print_function_call_with_float_as_argument) {
```

File: tests/backend/semantizer.cpp [hunk 4]
```diff
@@ -44,6 +44,76 @@ TEST(Semantizer, can_fill_variables_table) {
...
+                           "                  TypeName: FloatType\n"
+                           "                  BinaryOperation: Div\n"
+                           "                    BinaryOperation: Mult\n"
+                           "                      VariableName: x\n"
+                           "                      IntegerLiteralValue: 2\n"
+                           "                    IntegerLiteralValue: 1\n";
+    ASSERT_EQ(tree_str, tree.dump());
+}
+
+TEST(Semantizer, can_work_with_print_function_with_int_args) {
```
Comment (vla5924) [inline tests/backend/semantizer.cpp:83]:
```suggestion
TEST(Semantizer, can_process_print_function__call_with_int_as_argument) {
```

**Review by vla5924 — CHANGES_REQUESTED** [review]

**Review by arteboss — APPROVED** [review]
KPACABA
image

**Review by vla5924 — APPROVED** [review]

**Pull request merged**